model/project_event.go

```diff
@@ -247,7 +266,23 @@ func (p *ProjectSettings) resolveDefaults() *ProjectSettingsEvent {
 	return projectSettingsEvent
 }
 
+// LogProjectModified logs an event for a modification of a project's settings.
 func LogProjectModified(projectId, username string, before, after *ProjectSettings) error {
+	eventData := constructProjectChangeEvent(username, before, after)
+	if eventData == nil {
+		return nil
+	}
+	return LogProjectEvent(event.EventTypeProjectModified, projectId, *eventData)
+}
+
+// LogProjectRepoAttachment logs an event for either the attachment of a project to a repo,
+// or a detachment of a project from a repo.
+func LogProjectRepoAttachment(projectId, username, attachmentType string, before, after *ProjectSettings) error {
```

Comment:
Putting this as a random thread to make conversation easier:

@hadjri from the ticket description I think it is important that we have events for repos show up in the event log (not necessarily the PROJECT_ADDED variable since it's new and we weren't doing that before; do the repo attachment ones show up?

> By the time the button was clicked, the project had undergone changes and had event logs, but the repo was initialized with no event logs.

This confuses me a little because the ticket makes it seem like the repo itself was modified and was missing event logs. Does that not appear to be the case?

Reply:
So yes I'm pretty sure that the linked repo was never modified after its creation and what the ticket was initially implying didn't actually happen. A few requests to modify the repo have come in but all failed due to insufficient permissions by the requester. It's looking like regular repo modifications log events without issue.

I reproduced this in Spruce - attaching a project to a new repo initializes the repo with all of the variables from the attaching project. This results in the repo having custom variables set, making it seem like it was modified when it was just copied over.

Reply:
<img width="418" alt="image" src="https://github.com/evergreen-ci/evergreen/assets/19805673/f2ad7eb9-1c9c-411d-9f1c-c642357ad334">

<img width="410" alt="image" src="https://github.com/evergreen-ci/evergreen/assets/19805673/fa6c2103-508a-43e0-8ace-a1d135c300e7">

Attaching what the event log looks like for a project attaching to a repo and then subsequently detaching from it. First picture is the event logs for the project, second is for the repo. The project event logs contain logs saying it is attaching to / detaching from a repo, and the repo event logs only get a single event log for the initial attachment, i.e. creation

Reply:
Ah okay that makes sense. Thanks!